lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -309,6 +336,15 @@ defmodule KinesisClient.Stream.Shard.Producer do
 
         {:ok, messages, new_state}
 
+      {:ok, %{"ChildShards" => child_shards, "MillisBehindLatest" => _, "Records" => []}} ->
```

Comment:
This is the core missing clause: When we reach the end of consuming a shard, we get `ChildShards` back that point to the new shards to handle.

This version of these updates doesn't include handling these new child shards, as the coordinator does a lot of stuff to get that working. But, I've roughed in where we can handle it, and we may be able to extend it from there.

Reply:
I've since updated it to support adding child shards.